staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go

```diff
@@ -512,3 +515,78 @@ func TestHistogramVec_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestGetHistogramVecFromGatherer(t *testing.T) {
+	var registerMetrics sync.Once
+	tests := []struct {
+		name    string
+		lvMap   map[string]string
+		wantVec HistogramVec
+	}{
+		{
+			name:  "filter with one label",
+			lvMap: map[string]string{"label1": "value1-0"},
+			wantVec: HistogramVec{
+				&Histogram{&dto.Histogram{
+					SampleCount: uint64Ptr(1),
+					SampleSum:   pointer.Float64Ptr(1.5),
+					Bucket: []*dto.Bucket{
+						{CumulativeCount: uint64Ptr(0), UpperBound: pointer.Float64Ptr(0.5)},
+						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(5.0)},
+					},
+				}},
+				&Histogram{&dto.Histogram{
+					SampleCount: uint64Ptr(1),
+					SampleSum:   pointer.Float64Ptr(2.5),
+					Bucket: []*dto.Bucket{
+						{CumulativeCount: uint64Ptr(0), UpperBound: pointer.Float64Ptr(0.5)},
+						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(5.0)},
+					},
+				}},
+			},
+		},
+		{
+			name:  "filter with two labels",
+			lvMap: map[string]string{"label1": "value1-0", "label2": "value2-1"},
+			wantVec: HistogramVec{
+				&Histogram{&dto.Histogram{
+					SampleCount: uint64Ptr(1),
+					SampleSum:   pointer.Float64Ptr(2.5),
+					Bucket: []*dto.Bucket{
```

Comment:
nit: we don't really need to check the Buckets, as they don't change for the query with different label values.

So you can use `cmpopts.IgnoreFields` to simplify the test table.

Reply:
updated the `upperBounds`, so that the two `Histogram` will hold two different `Bucket` list, I'd like to keep the buckets to make sure the `Histogram` is built successfully as expected. I will remove them if you have strong opinion on this.